corehq/apps/api/resources/pagination.py

```diff
@@ -60,3 +65,31 @@ def page(self):
             self.collection_name: self.objects,
             'meta': meta,
         }
+
+
+def response_for_cursor_based_pagination(query, request_params, datasource_adapter):
+    """Creates a response dictionary that can be used for cursor based pagination
+     :returns: The response dictionary
+    """
+    records = get_datasource_records(query, datasource_adapter)
+    return {
+        "objects": records,
+        "meta": {
+            "next": _get_next_url_params(records),
```

Comment:
So "next" is a URL, and `_get_next_url_params()` only returns URL params.

But this function gets called by `corehq.apps.api.resources.v0_5.get_datasource_data()` and it has access to `request`, which can tell you the URL that the request came in on. That'll give you what you need to construct a full URL, with the request params for the next page, and also to preserve any other (future?) params. Maybe unnecessary now, but also maybe more robust than assuming that next URL params will always be the only params.

This should point you in the right direction: https://docs.djangoproject.com/en/3.2/ref/request-response/#django.http.HttpRequest.build_absolute_uri